update doc for template func (#74)
helmfile/helmfile: docs/templating_funcs.md

> # templating funcs

Reviewer: ```suggestion
# Template Functions
```

I think this casing and the wording is more idiomatic!

#### `requiredEnv` 
The `requiredEnv` function allows you to declare a particular environment variable as required for template rendering.
If the environment variable is unset or empty, the template rendering will fail with an error message.

```yaml
{{ $envValue := requiredEnv "envName" }}
```

#### `exec`

> The `exec` function allows you to run a command, then the stdout of command as the function output. when the command fails, the template rendering will fail with an error message.

Reviewer: ```suggestion
The `exec` function allows you to run a command, returning the stdout of the command. When the command fails, the template rendering will fail with an error message.
```

```yaml
{{ $cmdOutpot := exec "./mycmd" (list "arg1" "arg2" "--flag1") }}
```

#### `readFile`

> The `readFile` function allows you to read a file and return its content as the function output. when has failed, the template rendering will fail with an error message.

Reviewer: ```suggestion
The `readFile` function allows you to read a file and return its content as the function output. On failure, the template rendering will fail with an error message.
```

```

#### `fromYaml`
The `fromYaml` function allows you to convert a YAML string to a value. When has failed, the template rendering will fail with an error message.

```yaml
{{ $value :=  $yamlString | fromYaml }}
```

#### `setValueAtPath`
The `setValueAtPath` function allows you to set a value at a path. When has failed, the template rendering will fail with an error message.

```yaml
{{ $value | setValueAtPath "path.key" $newValue }}
```

#### `get`
The `get` function allows you to get a value at a path. when defaultValue not set. It will return nil. When has failed, the template rendering will fail with an error message.

```yaml
{{ $Getvalue :=  $value | get "path.key" "defaultValue" }}
```

#### `getOrNil`
The `getOrNil` function allows you to get a value at a path. when defaultValue not set. It will return nil. When has failed, the template rendering will fail with an error message.

```yaml
{{ $GetOrNlvalue :=  $value | getOrNil "path.key" }}
```

#### `tpl`
The `tpl` function allows you to render a template. When has failed, the template rendering will fail with an error message.

```yaml
{{ $tplValue :=  $value | tpl "{{ .Value.key }}" }}
```

#### `required`

> The `required` function allows you to check if a value is not empty and return it. When has failed, the template rendering will fail with an error message.

Reviewer: ```suggestion
The `required` function returns the argument as-is only if it is not empty. If empty, the template rendering will fail with an error message.
```

```yaml
{{ $requiredValue :=  $value | required "value not set" }}
```

#### `fetchSecretValue`

> The `fetchSecretValue` function allows you to fetch a secret value from Vault. When has failed, the template rendering will fail with an error message.

Reviewer: ```suggestion
The `fetchSecretValue` function parses the argument as a [vals](https://github.com/variantdev/vals) ref URL, retrieves and returns the remote secret value referred by the URL. In case it failed to access the remote secret backend for whatever reason or the URL was invalid, the template rendering will fail with an error message.
```

```yaml
{{ $fetchSecretValue :=  fetchSecretValue "secret/path" }}
```

### `expandSecretRefs`

> The `expandSecretRefs` function allows you to expand a secret reference. When has failed, the template rendering will fail with an error message.

Reviewer: ```suggestion
The `expandSecretRefs` function takes an object as the argument and expands every [vals](https://github.com/variantdev/vals) secret reference URL embedded in the object's values. See ["Remote Secrets" page in our documentation](./templating_funcs.md) for more information.
```
